pkgs/development/pharo/default.nix

```diff
@@ -1,101 +1,100 @@
-    cairo
-    libgit2
-    libpng
-    pixman
-    SDL2
-  ];
+    buildInputs = [
+      cairo
+      freetype
+      libffi
+      libgit2
+      libpng
+      libuuid
+      openssl
+      pixman
+      SDL2
+    ];
 
-  nativeBuildInputs = [
-    cmake
-    freetype
-    gcc
-    git
-    gnumake
-    libffi
-    libuuid
-    makeBinaryWrapper
-    openssl
-    pixman
-    SDL2
-    unzip
-  ];
+    nativeBuildInputs = [ cmake makeBinaryWrapper ];
 
-  cmakeFlags = [
-    # Necessary to perform the bootstrapping without already having Pharo available.
-    "-DGENERATED_SOURCE_DIR=."
-    "-DALWAYS_INTERACTIVE=ON"
-    "-DBUILD_IS_RELEASE=ON"
-    "-DGENERATE_SOURCES=OFF"
-    # Prevents CMake from trying to download stuff.
-    "-DBUILD_BUNDLE=OFF"
-  ];
+    dontPatchELF = true;
 
-  installPhase = ''
-    runHook preInstall
+    cmakeFlags = [
+      # Necessary to perform the bootstrapping without already having Pharo available.
+      "-DGENERATED_SOURCE_DIR=."
+      "-DALWAYS_INTERACTIVE=ON"
+      "-DBUILD_IS_RELEASE=ON"
+      "-DGENERATE_SOURCES=OFF"
+      # Prevents CMake from trying to download stuff.
+      "-DBUILD_BUNDLE=OFF"
+    ];
 
-    cmake --build . --target=install
-    mkdir -p "$out/lib"
-    mkdir "$out/bin"
-    cp build/vm/*.so* "$out/lib/"
-    cp build/vm/pharo "$out/bin/pharo"
-    patchelf --allowed-rpath-prefixes "$NIX_STORE" --shrink-rpath "$out/bin/pharo"
-    wrapProgram "$out/bin/pharo" --set LD_LIBRARY_PATH "${library_path}"
+    installPhase = ''
+      runHook preInstall
 
-    runHook postInstall
-  '';
+      cmake --build . --target=install
+      mkdir -p "$out/lib" "$out/bin"
+      cp build/vm/*.so* "$out/lib/"
+      cp build/vm/pharo "$out/bin/pharo"
 
-  meta = with lib; {
-    description = "Clean and innovative Smalltalk-inspired environment";
-    homepage = "https://pharo.org";
-    license = licenses.mit;
-    longDescription = ''
-      Pharo's goal is to deliver a clean, innovative, free open-source
-      Smalltalk-inspired environment. By providing a stable and small core
-      system, excellent dev tools, and maintained releases, Pharo is an
-      attractive platform to build and deploy mission critical applications.
+      runHook postInstall
     '';
-    maintainers = [ ];
-    platforms = lib.platforms.linux;
-  };
-}
+
+    preFixup = let
+      libPath = lib.makeLibraryPath (finalAttrs.buildInputs ++ [
+        stdenv.cc.cc.lib
+        "$out"
+      ]);
+    in ''
+      patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+        --set-rpath "${libPath}" \
+        $out/bin/pharo
+      patchelf --shrink-rpath $out/bin/pharo
+      patchelf --set-rpath "${libPath}" $out/lib/*.so
+      patchelf --shrink-rpath $out/lib/*.so
+
+      patchelf --set-rpath "${libPath}" $out/lib/libPharoVMCore.so
+      ln -s "${libgit2}/lib/libgit2.so" $out/lib/libgit2.so.1.1
+      wrapProgram "$out/bin/pharo" --argv0 $out/bin/pharo --set LD_LIBRARY_PATH "${libPath}"
```

Comment:
maybe this should append or prepend, in case someone attempts to use libsane with this programming language.